### PR TITLE
Surface Program class logging pattern

### LINKED ISSUE #15370: Add an example for providing a pre-built configuration to Host
As far as I can tell there is currently no example which shows how to:

1. Clear out the default configuration sources added by `CreateDefaultBuilder`.
2. Add a pre-built `IConfiguration` via `AddConfiguration`.

As raised in [this StackOverflow question](https://stackoverflow.com/questions/58530942/how-to-read-configuration-settings-before-initializing-a-host-in-asp-net-core/58594026).

If this is something that would be beneficial to the docs I'd be happy to open a PR.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0eb4d762-3910-2c57-8741-3306c15537b6
* Version Independent ID: ba2254a0-ea8e-9b28-e94b-edd0fc58c043
* Content: [Configuration in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.0#feedback)
* Content Source: [aspnetcore/fundamentals/configuration/index.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/fundamentals/configuration/index.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #15370

**_Highly prototyped_** for discussion/modification.

### CODE REVIEW

File: aspnetcore/fundamentals/logging/index.md [hunk 1]
```diff
@@ -125,6 +125,74 @@ To write logs in the `Program` class of an ASP.NET Core app, get an `ILogger` in
 
 [!code-csharp[](index/samples/3.x/TodoApiSample/Program.cs?name=snippet_LogFromMain&highlight=9,10)]
 
+Logging in `ConfigureServices` isn't directly supported. However, a static third-party logger can be used. In the following example, a [Serilog](https://serilog.net/) logger is used to log in `CreateHostBuilder`:
```
Comment (Tratcher) [inline aspnetcore/fundamentals/logging/index.md]:
```suggestion
Logging in `ConfigureServices` isn't directly supported. However, a separate logger can be used. In the following example, a [Serilog](https://serilog.net/) logger is used to log in `CreateHostBuilder`:
```

File: aspnetcore/fundamentals/logging/index.md [hunk 2]
```diff
@@ -125,6 +125,74 @@ To write logs in the `Program` class of an ASP.NET Core app, get an `ILogger` in
...
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+        }
+        catch (Exception ex)
+        {
+            Log.Fatal(ex, "Host builder error");
+
+            return null;
```
Comment (Tratcher) [inline aspnetcore/fundamentals/logging/index.md]: rethrow so Main doesn't have to check for null.
```suggestion
            throw;
```

File: aspnetcore/fundamentals/logging/index.md (hunk 1 above)
Comment (Tratcher) [reply aspnetcore/fundamentals/logging/index.md]: This description says you can't log in ConfigureServices, but the sample shows how to log in CreateHostBuilder? The sample should also log in ConfigureServices?

File: aspnetcore/fundamentals/logging/index.md [hunk 3]
```diff
@@ -125,6 +125,74 @@ To write logs in the `Program` class of an ASP.NET Core app, get an `ILogger` in
...
+                {
+                    services.AddRazorPages();
+                })
+                .ConfigureAppConfiguration((hostingContext, config) =>
+                {
+                    config.AddConfiguration(builtConfig);
+                })
+                .ConfigureLogging(logging =>
+                {
+                    logging.AddSerilog();
```
Comment (Tratcher) [inline aspnetcore/fundamentals/logging/index.md]:
```suggestion
                    // AddSerilog utilizes the static config specified above in `Log.Logger`.
                    logging.AddSerilog();
```

File: aspnetcore/fundamentals/logging/index.md (hunk 1 above)
Comment (guardrex) [reply aspnetcore/fundamentals/logging/index.md]: Maybe better as “logging while building the host” isn’t supported. Is that accurate?

Comment (guardrex) [reply aspnetcore/fundamentals/logging/index.md]: Like that? ☝️

File: aspnetcore/fundamentals/logging/index.md [hunk 4]
```diff
@@ -125,6 +125,69 @@ To write logs in the `Program` class of an ASP.NET Core app, get an `ILogger` in
 
 [!code-csharp[](index/samples/3.x/TodoApiSample/Program.cs?name=snippet_LogFromMain&highlight=9,10)]
 
+Logging during host construction isn't directly supported. However, a static third-party logger can be used. In the following example, a [Serilog](https://serilog.net/) logger is used to log in `CreateHostBuilder`. `AddSerilog` uses the static configuration specified in `Log.Logger`:
```
Comment (Tratcher) [inline aspnetcore/fundamentals/logging/index.md]:
```suggestion
Logging during host construction isn't directly supported. However, a separate logger can be used. In the following example, a [Serilog](https://serilog.net/) logger is used to log in `CreateHostBuilder`. `AddSerilog` uses the static configuration specified in `Log.Logger`:
```

**Pull request merged**